extensions/import/src/wizard/pages/summaryPage.ts

```diff
@@ -0,0 +1,165 @@
+
+		await this.view.initializeModel(this.form);
+		return true;
+	}
+
+	async onPageEnter(): Promise<boolean> {
+		this.loading.loading = true;
+		this.populateTable();
+		await this.handleImport();
+		this.loading.loading = false;
+		this.instance.setImportAnotherFileVisibility(true);
+
+		return true;
+	}
+
+	async onPageLeave(): Promise<boolean> {
+		this.instance.setImportAnotherFileVisibility(false);
+
+		return true;
+	}
+
+	private populateTable() {
+		this.table.updateProperties({
+			data: [
+				['Server name', this.model.server.providerName],
+				['Database name', this.model.database],
+				['Table name', this.model.table],
+				['Table schema', this.model.schema],
+				['File to be imported', this.model.filePath]],
+			columns: ['Object type', 'Name'],
+			width: 600,
+			height: 200
+		});
+	}
+
+	private async handleImport(): Promise<boolean> {
+		let changeColumnResults = [];
+		this.model.proseColumns.forEach((val, i, arr) => {
+			let columnChangeParams = {
+				index: i,
+				newName: val.columnName,
+				newDataType: val.dataType,
+				newNullable: val.nullable,
+				newInPrimaryKey: val.primaryKey
+			};
+			changeColumnResults.push(this.provider.sendChangeColumnSettingsRequest(columnChangeParams));
+		});
+
+		let result: InsertDataResponse;
+		let err;
+		try {
+			result = await this.provider.sendInsertDataRequest({
+				connectionString: await this.getConnectionString(),
+				//TODO check what SSMS uses as batch size
+				batchSize: 500
+			});
+		} catch (e) {
+			err = e.toString();
+		}
+
+		let updateText: string;
+		if (!result || !result.result.success) {
+			updateText = '✗ ';
+			if (!result) {
+				updateText += err;
+			} else {
+				updateText += result.result.errorMessage;
+			}
+		} else {
+			let rows = await this.getCountRowsInserted();
+			if (rows < 0) {
+				updateText = localize('flatFileImport.success.norows', '✔ Awesome! You have successfully inserted the data into a table.');
+			} else {
+				updateText = localize('flatFileImport.success.rows', '✔ Awesome! You have successfully inserted %rows% rows.').replace('%rows%', String(rows));
+			}
+		}
+		this.statusText.updateProperties({
+			value: updateText
+		});
+		return true;
+	}
+
+	/**
+	 * Gets the connection string to send to the middleware
+	 * @returns {Promise<string>}
+	 */
+	private async getConnectionString(): Promise<string> {
+		let options = this.model.server.options;
+		let connectionString: string;
+
+		if (options.authenticationType === 'Integrated') {
+			connectionString = `Data Source=${options.server + (options.port ? `,${options.port}` : '')};Initial Catalog=${this.model.database};Integrated Security=True`;
+		} else {
+			let credentials = await sqlops.connection.getCredentials(this.model.server.connectionId);
+			connectionString = `Data Source=${options.server + (options.port ? `,${options.port}` : '')};Initial Catalog=${this.model.database};Integrated Security=False;User Id=${options.user};Password=${credentials.password}`;
```

Comment:
I think Karl has added the connection string API now so we should use it instead of these manually-constructed connection strings we used for the hackathon

Reply:
What is the code to use this?

Reply:
Merge in the latest version of master and then search `sqlops.d.ts` for the `getConnectionString` method

Reply:
Done!